### PR TITLE
CATTY-439 Implement SetLookByIndexBrick

### PR DESCRIPTION
**Pull request by zwerg4**
Implemented a new Brick called "SetLookByIndexBrick". This Brick changes the Look of the current SpriteObject to a look identified by a given number.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [X] Include the name of the Jira ticket in the PR’s title
- [X] Verify that the Jira ticket is in the status *Ready for Development*
- [X] Include a summary of the changes plus the relevant context
- [X] Choose the proper base branch (*develop*)
- [X] Confirm that the changes follow the project’s coding guidelines
- [X] Verify that the changes generate no compiler or linter warnings
- [X] Perform a self-review of the changes
- [X] Verify to commit no other files than the intentionally changed ones
- [X] Include reasonable and readable tests verifying the added or changed behavior
- [X] Confirm that new and existing unit tests pass locally
- [X] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [X] Stick to the project’s git workflow (rebase and squash your commits)
- [X] Verify that your changes do not have any conflicts with the base branch
- [X] After the PR, verify that all CI checks have passed
- [X] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: src/CattyTests/Bricks/SetLookByIndexBrickTests.swift [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        spriteObject.spriteNode.currentLook = look
+        spriteObject.spriteNode.currentUIImageLook = UIImage(contentsOfFile: filePath!)
+    }
+
+    func testSetLookByValidIndex() {
+        brick.lookIndex = Formula(integer: 3)
+
+        var action = brick.actionBlock(formulaInterpreter)
+        action()
+        XCTAssertEqual(spriteNode.currentLook, look3, "setLookByIndexBrick not correct")
```
Comment (aeluna) [inline src/CattyTests/Bricks/SetLookByIndexBrickTests.swift:84]: What do you mean with "setLookByIndexBrick not correct"? I think this is rather confusing, therefore I would suggest, to either remove the String or to rephrase it. And please do so for all the other XCTAssertEquals in this file.

**Review by aeluna — CHANGES_REQUESTED** [review]

**Review by aeluna — APPROVED** [review]

File: src/CattyTests/Resources/ParserTests/Custom/ValidProjectAllBricks/ValidProjectAllBricks0991.xml [hunk 2]
```diff
@@ -247,6 +247,15 @@
                       <brick type="NextLookBrick">
                           <commentedOut>false</commentedOut>
                       </brick>
+                      <brick type="SetLookByIndexBrick">
```
Comment (m-herold) [inline src/CattyTests/Resources/ParserTests/Custom/ValidProjectAllBricks/ValidProjectAllBricks0991.xml:250]: Please also add Brick to `ValidProjectAllBricks0992.xml` and `ValidProjectAllBricks0993.xml`.

File: src/Catty/DataModel/Bricks/Look/SetLookByIndexBrick.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    override func setDefaultValuesFor(_ spriteObject: SpriteObject!) {
+        self.lookIndex = Formula(integer: kBackgroundObjects)
+    }
+
+    func allowsStringFormula() -> Bool {
+        false
+    }
+
+    override func isDisabledForBackground() -> Bool {
+        false
```
Comment (m-herold) [inline src/Catty/DataModel/Bricks/Look/SetLookByIndexBrick.swift:70]: Please disable this Brick for "background objects".

File: src/Catty/Setup/CatrobatSetup+Bricks.swift [hunk 4]
```diff
@@ -88,6 +88,7 @@
             ThinkForBubbleBrick(),
             ThinkBubbleBrick(),
             AskBrick(),
+            SetLookByIndexBrick(),
```
Comment (m-herold) [inline src/Catty/Setup/CatrobatSetup+Bricks.swift:91]: Please order this Brick below the "Switch to look" Brick.

File: src/Catty/SetLookByIndexBrick+Instruction.swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            if image == nil {
+                print("LoadImageFromDisk")
+                cache?.loadImageFromDisk(withPath: self.pathForLook(look: lookAtIndex))
+                if let look = self.pathForLook(look: lookAtIndex) {
+                    guard let imageFromDisk = UIImage(contentsOfFile: look) else { return }
+                    image = imageFromDisk
+                }
+            }
+
+            spriteNode.currentLook = lookAtIndex
```
Comment (m-herold) [inline src/Catty/SetLookByIndexBrick+Instruction.swift:56]: The image could theoretically still be `nil` at this point. To be on the safe side, please do not run `executeFilter` and change the `currentLook` in that case (please also add a test to verify this scenario having an invalid look path).

**Review by m-herold — CHANGES_REQUESTED** [review]

File: src/Catty/Setup/CatrobatSetup+Bricks.swift (hunk 4 above)
Comment (zwerg4) [reply src/Catty/Setup/CatrobatSetup+Bricks.swift]: could not find "Switch to look" ordered it under "PreviousLookBrick"

Comment (m-herold) [reply src/Catty/Setup/CatrobatSetup+Bricks.swift]: This is the second Brick in this category, internally referred to as "SetLookBrick".

**Review by m-herold — APPROVED** [review]

**Comment by m-herold** [comment]
Works perfectly, thank you!

**Pull request merged**